src/field_take_photo.c

```diff
@@ -0,0 +1,832 @@
+            sub_0205F484(taskData->parent->mapObjects[1]);
+            LocalMapObject *mapObject = taskData->parent->mapObjects[1];
+            if (ov01_0220553C(mapObject)) {
+                ov01_021F902C(1, mapObject);
+            }
+        }
+        FreeToHeap(taskData);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static LocalMapObject *createPartymonMapObject(MapObjectManager *objectMan, u16 species, u16 form, int gender, int direction, int x, int y, BOOL shiny) {
+    LocalMapObject *ret = createSpecialMapObject(objectMan, FollowMon_GetSpriteID(species, form, gender), direction, x, y, 253);
+    MapObject_SetParam(ret, 0, 2);
+    FollowMon_SetObjectParams(ret, species, form, shiny);
+    return ret;
+}
+
+static LocalMapObject *createSpecialMapObject(MapObjectManager *objectMan, int spriteId, int direction, int x, int y, int localId) {
+    LocalMapObject *ret = CreateSpecialFieldObject(objectMan, x, y, direction, spriteId, 0, 1);
+    GF_ASSERT(ret != NULL);
+    MapObject_SetID(ret, localId);
+    MapObject_SetType(ret, 0);
+    MapObject_SetFlagID(ret, 0);
+    MapObject_SetXRange(ret, -1);
+    MapObject_SetYRange(ret, -1);
+    MapObject_SetFlagsBits(ret, MAPOBJECTFLAG_UNK10 | MAPOBJECTFLAG_UNK13);
+    MapObject_ClearFlagsBits(ret, MAPOBJECTFLAG_UNK7 | MAPOBJECTFLAG_UNK8);
+    MapObject_SetFlag29(ret, TRUE);
+    return ret;
+}
+
+// this function was partially stubbed, or called a static inline child routine that was subbed
+static void sub_0206AF78(FieldSystem *fieldSystem, PHOTO *photo) {
+    int index = 0;
+    LocalMapObject *curObj;
+    PlayerAvatar_GetMapObject(fieldSystem->playerAvatar);
+    while (MapObjectManager_IterObjects(fieldSystem->mapObjectManager, &curObj, &index, MAPOBJECTFLAG_ACTIVE)) {
+    }
+}
+
+static void FieldTakePhoto_RemoveAllMapObjects(FieldTakePhoto *takePhoto) {
+    for (int i = 0; i < takePhoto->numObjects; ++i) {
+        if (takePhoto->mapObjects[i] != NULL) {
+            MapObject_Remove(takePhoto->mapObjects[i]);
+        }
+    }
+}
+
+static void setCameraParam(Camera *camera) {
+    Camera_SetDistance(sCameraParam.distance, camera);
+    Camera_SetAnglePos(&sCameraParam.angle, camera);
+    Camera_SetPerspectiveAngle(sCameraParam.perspective, camera);
+    Camera_ApplyPerspectiveType(sCameraParam.perspectiveType, camera);
+    Camera_OffsetLookAtPosAndTarget(&sCameraParam.lookAt, camera);
+}
+
+static void Photo_InitFromArcData(PHOTO *photo, FieldSystem *fieldSystem, u8 iconId, int mapId, int x, int y, int a6, int a7, u16 a8, u16 subjectObjId) {
+    int partySize;
+    SaveData *saveData = FieldSystem_GetSaveData(fieldSystem);
+    PlayerProfile *profile = Save_PlayerData_GetProfileAddr(saveData);
+    Party *party = SaveArray_Party_Get(saveData);
+    int i;
+
+    Photo_Init(photo);
+
+    RTCDate date;
+    GF_RTC_CopyDate(&date);
+    photo->date = ((date.year & 0xFF) << 24) | ((date.month & 0xFF) << 16) | ((date.day & 0xFF) << 8) | date.week;
```

Comment:
do we have a macro/inline func for this?

Reply:
do we... need one?

Reply:
I mean we don't *need* one, it'd just be a nice qol thing to have cause it's repeated more than once

Reply:
not gonna bother tbh